### ISSUE TITLE
Installing a bundle package on remote machine

### ISSUE THREAD

**Issue by nirbar** [wixtoolset/issues#5775]
# Feature request

Enable bootstrapper to install a package on remote unattended machine. 
The feature will enable multi-machine systems where some of the machines are headless to be installed from a single machine.

## Describe the scenario and benefits that the feature supports.

I have a requirement to install a server-client system where client machines are headless. The bootstrapper is required to install the server packages locally and the client packages on client machines over local network. 

## Describe how you're accomplishing the feature today (if possible).

Today server packages are installed with WiX bootstrapper.
Client package is installed by connecting remotely with a terminal and running a silent install.

## Describe what you'd like the new feature to do.

I'd like to add support for RemoteBootstrapper elements roughly looking:
~~~~~~~
<RemoteBootstrapper SourceFile="PackagePath" 
          IP="[RemoteMachineIP]" 
          User="[UserName]" 
          Password="[***********]"
          InstallCondition="SomeCondition"
          InstallCommand="Var1=Val1 Var2=[Val2]"/>
~~~~~~~

The bootstrapper will be able to detect, plan and apply on the remote machine.

## Design Thoughts

- During detect phase, the bootstrapper will establish a connection to the remote machine to run detection logic on it. That means that BA will need to present UI before starting package detection to get machine IP and credentials from the operator.
- Connection to the remote machine can be developed by incorporating [PAExec](https://www.poweradmin.com/paexec/)-based code in burn engine
- The local engine will launch the remote engine as an embedded bootstrapper. That means the remote engine will be very similar to existing engine except that the communication pipe will be connected to the host machine.
- Remote engine will send detection results to the host
- Remote apply results will be interpreted properly so that reboot request will launch reboot on the remote machine rather than the local.
- Network administrator is expected to define permissions on the client machines so that the user who performs the install on server machines has admin rights on the remote clients.

**Comment by barnson** [comment]
Out of scope for Burn itself. Really belongs in a higher layer that can handle more than just bundles.
